cmd/telegraf/telegraf.go

```diff
@@ -3,138 +3,78 @@ package main
-	"filter the outputs to enable, separator is :")
-var fOutputList = flag.Bool("output-list", false,
-	"print available output plugins.")
-var fAggregatorFilters = flag.String("aggregator-filter", "",
-	"filter the aggregators to enable, separator is :")
-var fProcessorFilters = flag.String("processor-filter", "",
-	"filter the processors to enable, separator is :")
-var fUsage = flag.String("usage", "",
-	"print usage for a plugin, ie, 'telegraf --usage mysql'")
-
-// Initialize the subcommand `telegraf config`
-// This duplicates the above filters which are used for `telegraf --sample-config` and `telegraf --deprecation-list`
-var configCmd = flag.NewFlagSet("config", flag.ExitOnError)
-var fSubSectionFilters = configCmd.String("section-filter", "",
-	"filter the sections to print, separator is ':'. Valid values are 'agent', 'global_tags', 'outputs', 'processors', 'aggregators' and 'inputs'")
-var fSubInputFilters = configCmd.String("input-filter", "",
-	"filter the inputs to enable, separator is :")
-var fSubOutputFilters = configCmd.String("output-filter", "",
-	"filter the outputs to enable, separator is :")
-var fsubAggregatorFilters = configCmd.String("aggregator-filter", "",
-	"filter the aggregators to enable, separator is :")
-var fSubProcessorFilters = configCmd.String("processor-filter", "",
-	"filter the processors to enable, separator is :")
-
-//nolint:varcheck,unused // False positive - this var is used for non-default build tag: windows
-var fService = flag.String("service", "",
-	"operate on the service (windows only)")
-
-//nolint:varcheck,unused // False positive - this var is used for non-default build tag: windows
-var fServiceName = flag.String("service-name", "telegraf",
-	"service name (windows only)")
-
-//nolint:varcheck,unused // False positive - this var is used for non-default build tag: windows
-var fServiceDisplayName = flag.String("service-display-name", "Telegraf Data Collector Service",
-	"service display name (windows only)")
-
-//nolint:varcheck,unused // False positive - this var is used for non-default build tag: windows
-var fServiceAutoRestart = flag.Bool("service-auto-restart", false,
-	"auto restart service on failure (windows only)")
-
-//nolint:varcheck,unused // False positive - this var is used for non-default build tag: windows
-var fServiceRestartDelay = flag.String("service-restart-delay", "5m",
-	"delay before service auto restart, default is 5m (windows only)")
-
-//nolint:varcheck,unused // False positive - this var is used for non-default build tag: windows
-var fRunAsConsole = flag.Bool("console", false,
-	"run as console application (windows only)")
-var fPlugins = flag.String("plugin-directory", "",
-	"path to directory containing external plugins")
-var fRunOnce = flag.Bool("once", false, "run one gather and exit")
+type App interface {
+	Init(<-chan error, Filters, GlobalFlags, WindowFlags)
+	Run() error
+}
 
-var stop chan struct{}
+type Telegraf struct {
```

Comment:
It makes a lot of sense to have a struct that owns the vars it uses instead of them all being globals!